lib/Dancer2/Core/Request.pm

```diff
@@ -199,6 +199,10 @@ sub deserialize {
         $serializer_fail = $_[1];
         $serializer_log_cb->(@_);
     };
+    # work-around to resolve a chicken-and-egg issue when instantiating a
+    # request object; the serializer needs that request object to deserialize
+    # the body params.
+    $self->serializer->has_request || $self->serializer->set_request($self);
```

Comment:
Request has a serializer, and serializer has the request.

The request attribute in Dancer2::Core::Role::Engine isn't a weak_ref.

Could that cause a memory leak?

Reply:
Possible but unlikely; the code in `Dancer2::Core::App->dispatch()` is where the request object is created then sets the request object into each defined engine without weakening the request object ref; that has no known memory leak.

Line 205 is above pre-empting a call to `$self->set_request($request, $defined_engines);` ( currently line 1444 of Dancer2::Core::App ) to resolve the chicken-and-egg issue.

Reply:
`Dancer2::Core::App` weakens a copy when it creates the engines. Check `_build_serializer_engine`, `_build_session_engine`, and `_build_template_engine`. You will find it there.

Reply:
We were already setting the request in the serializer instance as part of the app's dispatch, which hadn't caused any known problems, hence I wasn't concerned. Pretty trivial to weaken it to ensure it never bites us though.

Added commit 57834d3 to weaken the request object in *both* places is was being set in any engine instance.